crates/supervisor/storage/src/chaindb.rs

```diff
@@ -370,9 +370,42 @@ impl HeadRefStorageWriter for ChainDb {
 }
 
 impl Rewinder for ChainDb {
-    fn rewind_log_storage(&self, from_block: u64) -> Result<(), StorageError> {
+    fn rewind_log_storage(&self, to: &BlockInfo) -> Result<(), StorageError> {
```

Comment:
I think we don't need to depend on full `BlockInfo` here. We can just use the `BlockNumHash` like we are doing in `latest_derived_block_at_source` or `derived_to_source`.

Reply:
updated to BlockNumHash all places

Reply:
Yep, I understand that we are working with the BlockInfo in the reorg scenarios. But I think it's better to limit the information that required for an operation.